workers/common/src/main/kotlin/common/env/GitCredentialsGenerator.kt

```diff
@@ -0,0 +1,101 @@
+ * produce such a file, together with a Git configuration file that references it. It only processes infrastructure
+ * services whose credentials types contain [CredentialsType.GIT].
+ */
+class GitCredentialsGenerator : EnvironmentConfigGenerator<EnvironmentServiceDefinition> {
+    companion object {
+        /** The name of the file storing the actual credentials. */
+        private const val GIT_CREDENTIALS_FILE = ".git-credentials"
+
+        /** The name of the file with the Git configuration. */
+        private const val GIT_CONFIG_FILE = ".gitconfig"
+
+        private val logger = LoggerFactory.getLogger(GitCredentialsGenerator::class.java)
+
+        /**
+         * Return a string with the URL of this [InfrastructureService] with the credentials embedded as needed within
+         * the _.git-credentials_ file. Use [builder] to obtain secret references. Return *null* if the URL is invalid.
+         */
+        private fun InfrastructureService.urlWithCredentials(builder: ConfigFileBuilder): String? = runCatching {
+            buildString {
+                val serviceUrl = URI(url).toURL()
+                append(serviceUrl.protocol)
+                append("://")
+                append(builder.secretRef(usernameSecret)).append(':')
+                append(builder.secretRef(passwordSecret)).append('@')
+                append(serviceUrl.authority)
+                append(serviceUrl.path)
+            }
+        }.onFailure {
+            logger.error("Invalid URL for service '{}'. Ignoring it.", this, it)
+        }.getOrNull()
+    }
+
+    override val environmentDefinitionType: Class<EnvironmentServiceDefinition>
+        get() = EnvironmentServiceDefinition::class.java
+
+    override suspend fun generate(builder: ConfigFileBuilder, definitions: Collection<EnvironmentServiceDefinition>) {
+        definitions.filter { CredentialsType.GIT in it.credentialsTypes() }.takeUnless { it.isEmpty() }?.let {
+            generateGitCredentials(builder, it)
+            generateGitConfig(builder)
+        }
+    }
+
+    /**
+     * Generate the content of the _.git-credentials_ file using the given [builder] for the given [definitions].
+     */
+    private suspend fun generateGitCredentials(
+        builder: ConfigFileBuilder,
+        definitions: Collection<EnvironmentServiceDefinition>
+    ) {
+        builder.buildInUserHome(GIT_CREDENTIALS_FILE) {
+            definitions.mapNotNull { it.service.urlWithCredentials(builder) }
+                .forEach(this::println)
+        }
+    }
+
+    /**
+     * Generate the (static) content of the _.gitconfig_ file using the given [builder]. In this file, only the
+     * credential store is enabled.
+     */
+    private suspend fun generateGitConfig(builder: ConfigFileBuilder) {
+        builder.buildInUserHome(GIT_CONFIG_FILE) {
+            println("[credential]")
+            println("\thelper = store")
```

Comment:
When I look in my `.gitconfig` file there is:
`helper = store --file ~/.git-credentials`

Should we put this `--file` ?

Reply:
`~/.git-credentials` is the standard location. I would therefore only add the parameter if we - for whatever reasons - deviate from this standard.